etna/models/deadline_ma.py

```diff
@@ -0,0 +1,164 @@
+        """
+        Fit DeadlineMovingAverageModel model.
+        Parameters
+        ----------
+        df: pd.DataFrame
+            Data to fit on
+        regressors:
+            List of the columns with regressors(ignored in this model)
+        Raises
+        ------
+        ValueError
+            If freq of dataframe is not supported
+            If series is too short for chosen shift value
+
+        Returns
+        -------
+        :
+            Fitted model
+        """
+        freq = pd.infer_freq(df["timestamp"])
+        if freq not in self.freqs_available:
+            raise ValueError(f"{freq} is not supported! Use daily or hourly frequency!")
+
+        if set(df.columns) != {"timestamp", "target"}:
+            warnings.warn(
+                message=f"{type(self).__name__} does not work with any exogenous series or features. "
+                f"It uses only target series for predict/\n "
+            )
+        targets = df["target"]
+        timestamps = df["timestamp"]
+
+        if self.seasonality == SeasonalityMode.month:
+            first_index = timestamps.iloc[-1] - pd.DateOffset(months=self.window)
+
+        elif self.seasonality == SeasonalityMode.year:
+            first_index = timestamps.iloc[-1] - pd.DateOffset(years=self.window)
+
+        if first_index < timestamps.iloc[0]:
+            raise ValueError(
+                "Given series is too short for chosen shift value. Try lower shift value, or give" "longer series."
+            )
+
+        self.series = targets.loc[timestamps >= first_index]
+        self.timestamps = timestamps.loc[timestamps >= first_index]
+        self.shift = len(self.series)
+
+        return self
+
+    def predict(self, df: pd.DataFrame) -> np.ndarray:
+        """
+        Compute predictions from a DeadlineMovingAverageModel.
+        Parameters
+        ----------
+        df: pd.DataFrame
+            Used only for getting the horizon of forecast and timestamps
+        Returns
+        -------
+        :
+            Array with predictions.
+        """
+        timestamps = df["timestamp"]
+        index = pd.date_range(start=self.timestamps.iloc[0], end=timestamps.iloc[-1])
+        res = np.append(self.series.values, np.zeros(len(df)))
+        res = pd.DataFrame(res)
+        res.index = index
+        for i in range(len(self.series), len(res)):
+            for w in range(1, self.window + 1):
+                if self.seasonality == SeasonalityMode.month:
+                    prev_date = res.index[i] - pd.DateOffset(months=w)
+
+                elif self.seasonality == SeasonalityMode.year:
+                    prev_date = res.index[i] - pd.DateOffset(years=w)
+                if prev_date <= self.timestamps.iloc[-1]:
+                    res.loc[index[i]] += self.series.loc[self.timestamps == prev_date].values
+                else:
+                    res.loc[index[i]] += res.loc[prev_date].values
```

Comment:
could you please add a test for this "else" behavior?

smth like
```
pipeline = Pipeline(model=DeadlineMovingAverageModel(window=5, seasonality="month"), transforms=[], horizon=180)
```
and check that everything works